mxresolv/mxresolv.go

```diff
@@ -0,0 +1,145 @@
+package mxresolv
+
+import (
+	"context"
+	"math/rand"
+	"net"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/mailgun/holster/v4/clock"
+	"github.com/mailgun/holster/v4/collections"
+	"github.com/pkg/errors"
+	"golang.org/x/net/idna"
+)
+
+const (
+	cacheSize = 1000
+	cacheTTL  = 10 * clock.Minute
+)
+
+var (
+	errNullMXRecord   = errors.New("domain accepts no mail")
+	errNoValidMXHosts = errors.New("no valid MX hosts")
+
+	lookupResultCache *collections.LRUCache
```

Comment:
Global state - is bad. https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html